src/app/shared/filter-search.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+export type ItemsPerPage = 5 | 10 | 25 | 50 | undefined;
+export const ITEMS_PER_PAGE: ItemsPerPage[] = [5, 10, 25, 50, undefined];
+
+// tslint:disable-next-line:interface-over-type-literal
+export type Filter = {
+  field: string;
+  sort: 1 | -1;
+};
+
+@Injectable()
+export class FilterSearchService {
+  private filters$: BehaviorSubject<Filter[]> = new BehaviorSubject<Filter[]>(
+    []
+  );
+  private selectedFilter$: BehaviorSubject<Filter> = new BehaviorSubject<
+    Filter
+  >(null);
+  private searchText$: BehaviorSubject<string> = new BehaviorSubject<string>(
+    ''
+  );
+  private itemsPerPage$: BehaviorSubject<ItemsPerPage> = new BehaviorSubject<
+    ItemsPerPage
+  >(ITEMS_PER_PAGE[1]);
+
+  private statusFilter$: BehaviorSubject<string[]> = new BehaviorSubject<string[]>(
+    []
+  );
+
+   private selectedStatusFilter$: BehaviorSubject<string[]> = new BehaviorSubject<
+  string[]
+  >(null);
+  constructor() {}
+
+  public getFilters(): BehaviorSubject<Filter[]> {
```

Comment:
Make sure all of these copied functions are being used and provide JSDoc for each.

Reply:
This entire service is not being used anywhere. Can we remove it?

@gshaw1997 @nrwinner maybe you have context?

Reply:
I don't have any context, I don't think I've ever seen that file. Could it be a remnant of the old admin dashboard @gshaw1997 ? Regardless, its purpose appears to be to allow the sharing of filters between components. This is an anti pattern considering we should be segregating filters to the components they apply to, I'm game to remove it.